cores/esp8266/WString.cpp

```diff
@@ -342,10 +328,8 @@ unsigned char String::concat(const char *cstr) {
 }
 
 unsigned char String::concat(char c) {
-    char buf[2];
-    buf[0] = c;
-    buf[1] = 0;
-    return concat(buf, 1);
+    uint16_t buf = __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__ ? (uint16_t)c : (uint16_t)c << 8;
```

Comment:
Duplicate, please refactor

Reply:
I think also that most compilers perform this optimisation automatically. there might not be an observable behavior difference between old and new code.